app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.kt

```diff
@@ -194,7 +194,7 @@ class ZimHostActivity : BaseActivity(), ZimHostCallbacks, ZimHostContract.View {
 
   override fun onResume() {
     super.onResume()
-    presenter.loadBooks(sharedPreferenceUtil.hostedBooks)
+    sharedPreferenceUtil.hostedBooks?.let(presenter::loadBooks)
```

Comment:
this won't be necessary after you have updated `hostedBooks` to not return nullable values